resin/models/application.py

```diff
@@ -311,3 +313,150 @@ def disable_rolling_updates(self, app_id):
             'application', 'PATCH', params=params, data=data,
             endpoint=self.settings.get('pine_endpoint')
         )
+
+    def enable_device_urls(self, app_id):
+        """
+        Enable device urls for all devices that belong to an application
+
+        Args:
+            app_id (str): application id.
+
+        Returns:
+            OK/error.
+
+        Examples:
+            >> > resin.models.application.enable_device_urls('5685')
+            'OK'
+
+        """
+
+        params = {
+            'filter': 'belongs_to__application',
+            'eq': app_id
+        }
+        data = {
+            'is_web_accessible': True
+        }
+
+        return self.base_request.request(
+            'device', 'PATCH', params=params, data=data,
+            endpoint=self.settings.get('pine_endpoint')
+        )
+
+    def disable_device_urls(self, app_id):
+        """
+        Disable device urls for all devices that belong to an application.
+
+        Args:
+            app_id (str): application id.
+
+        Returns:
+            OK/error.
+
+        Examples:
+            >> > resin.models.application.disable_device_urls('5685')
+            'OK'
+
+        """
+
+        params = {
+            'filter': 'belongs_to__application',
+            'eq': app_id
+        }
+        data = {
+            'is_web_accessible': False
+        }
+
+        return self.base_request.request(
+            'device', 'PATCH', params=params, data=data,
+            endpoint=self.settings.get('pine_endpoint')
+        )
+
+    def grant_support_access(self, app_id, expiry_timestamp):
+        """
+        Grant support access to an application until a specified time.
+
+        Args:
+            app_id (str): application id.
+            expiry_timestamp (int): a timestamp in ms for when the support access will expire.
+
+        Returns:
+            OK/error.
+
+        Examples:
+            >> > resin.models.application.grant_support_access('5685', 1511974999000)
+            'OK'
+
+        """
+
+        if not expiry_timestamp or expiry_timestamp <= int((datetime.utcnow() - datetime.utcfromtimestamp(0)).total_seconds() * 1000):
+            raise exceptions.InvalidParameter('expiry_timestamp', expiry_timestamp)
+
+        params = {
+            'filter': 'id',
+            'eq': app_id
+        }
+
+        data = {
+            'is_accessible_by_support_until__date': expiry_timestamp
+        }
+
+        return self.base_request.request(
+            'application', 'PATCH', params=params, data=data,
+            endpoint=self.settings.get('pine_endpoint')
+        )
+
+    def revoke_support_access(self, app_id):
```

Comment:
I know nothing about the resin API, but just out of curiosity are there functions you can use to check if support access is enabled for a particular application or device?

Reply:
@lurch, we don't have an endpoint for it but we can check `support_expiry_date` property in every device to see whether it allows support access or not. We don't provide this function in both SDK atm but it's definitely a good point. Created an issue for it https://github.com/resin-io/resin-sdk-python/issues/54, I will implement it later.